crates/polars-lazy/src/frame/mod.rs

```diff
@@ -691,10 +691,24 @@ impl LazyFrame {
     ///       .collect()
     /// }
     /// ```
-    pub fn collect(self) -> PolarsResult<DataFrame> {
+    #[allow(unused_mut)]
+    pub fn collect(mut self) -> PolarsResult<DataFrame> {
         #[cfg(feature = "new_streaming")]
         {
             let force_new_streaming = self.opt_state.contains(OptState::NEW_STREAMING);
+            let auto_new_streaming = std::env::var("POLARS_AUTO_NEW_STREAMING")
+                .as_deref()
+                .unwrap_or("")
+                == "1";
+
+            // Turn off CSE if we go into new streaming engine.
```

Comment:
This check should be done in `optimize_with_scratch` to make sure it always happens. We should only set the `NEW_STREAMING` flag here, any optimization implications should happen somewhere else IMO.